src/common/selectors/entities.ts

```diff
@@ -10,6 +11,24 @@ export const getBuildingImages = createSelector(
 	images => images.filter((item: any) => item.build.status === 'building'),
 );
 
+export const getFilteredImages = createSelector(
+	getImages,
+	getImagesTextFilter,
+	(images, substr) => {
+		if (images.length && substr) {
+			return images.filter((image: any) => {
+				if (
```

Comment:
filter works with predicates, you don't need to use extra if's. Secondly, use a String.prototype.includes() instead image.distro.full_name.toLowerCase().indexOf(substr.toLowerCase()) !== -1